src/reduxCycles.js

```diff
@@ -0,0 +1,64 @@
+import xs from 'xstream';
+import {adapt} from '@cycle/run/lib/adapt';
+
+let actionListener = null;
+let stateListener = null;
```

Comment:
This means cycles will effectively be singleton, and these are coupled to `store` in some way anyway.

 I'd argue not the best way.

What if the middleware function returned from `createCycleMiddleware` had `makeActionDriver` and `makeStateDriver` added to that function? So the use would be:

```javascript
  const cycleMiddleware = createCycleMiddleware();
  const store = createStore(
    rootReducer,
    applyMiddleware(cycleMiddleware)
  );

  r(main, {
    ACTION: cycleMiddleware.makeActionDriver(),
    STATE: cycleMiddleware.makeStateDriver()
  })
```

You wouldn't have to pass the store, and scoping the listeners inside `createCycleMiddleware` means we're not singleton anymore, meaning several different cycle middleware can exist at the same time.